crates/stages/src/stages/hashing_account.rs

```diff
@@ -155,96 +159,50 @@ impl<DB: Database> Stage<DB> for AccountHashingStage {
         // genesis accounts are not in changeset.
         if to_block - from_block > self.clean_threshold || from_block == 1 {
             let tx = provider.tx_ref();
-            let stage_checkpoint = input
-                .checkpoint
-                .and_then(|checkpoint| checkpoint.account_hashing_stage_checkpoint());
-
-            let start_address = match stage_checkpoint {
-                Some(AccountHashingCheckpoint { address: address @ Some(_), block_range: CheckpointBlockRange { from, to }, .. })
-                    // Checkpoint is only valid if the range of transitions didn't change.
-                    // An already hashed account may have been changed with the new range,
-                    // and therefore should be hashed again.
-                    if from == from_block && to == to_block =>
-                {
-                    debug!(target: "sync::stages::account_hashing::exec", checkpoint = ?stage_checkpoint, "Continuing inner account hashing checkpoint");
-
-                    address
-                }
-                _ => {
-                    // clear table, load all accounts and hash it
-                    tx.clear::<tables::HashedAccounts>()?;
 
-                    None
-                }
+            // clear table, load all accounts and hash it
+            tx.clear::<tables::HashedAccounts>()?;
+
+            let mut accounts_cursor = tx.cursor_read::<RawTable<tables::PlainAccountState>>()?;
+
+            // channels used to return result of account hashing
+            let mut channels = Vec::new();
+            for chunk in &accounts_cursor.walk(None)?.chunks(100) {
+                // An _unordered_ channel to receive results from a rayon job
+                let (tx, rx) = mpsc::channel();
+                channels.push(rx);
+
+                let chunk = chunk.collect::<Result<Vec<_>, _>>()?;
+                // Spawn the hashing task onto the global rayon pool
+                rayon::spawn(move || {
```

Comment:
this has the same issue as sender recovery stage,

what we could do instead is add a pool that has < available_parallelism() threads

this would also solve the sender recovery issue I believe, 

ref https://github.com/paradigmxyz/reth/blob/273f3c734c0042c3776eddf6b796df884a18e7f7/crates/rpc/rpc-builder/src/constants.rs#L30-L31

Reply:
agree @joshieDo

Reply:
https://github.com/paradigmxyz/reth/pull/7267

should take care of it